packages/ruleset/test/no-ref-in-example.test.js

```diff
@@ -0,0 +1,202 @@
+      expect(results).toHaveLength(1);
+      for (const i in results) {
+        expect(results[i].code).toBe(ruleId);
+        expect(results[i].message).toBe(expectedMsg);
+        expect(results[i].severity).toBe(expectedSeverity);
+        expect(results[i].path.join('.')).toBe(expectedPaths[i]);
+      }
+    });
+
+    it('response example with $ref', async () => {
+      const testDocument = makeCopy(rootDocument);
+
+      testDocument.paths['/v1/drinks'].post.responses['201'].content[
+        'application/json'
+      ].example = {
+        $ref: '#/components/schemas/Error',
+      };
+
+      const expectedPaths = [
+        'paths./v1/drinks.post.responses.201.content.application/json.example.$ref',
+      ];
+
+      const results = await testRule(ruleId, rule, testDocument);
+      expect(results).toHaveLength(1);
+      for (const i in results) {
+        expect(results[i].code).toBe(ruleId);
+        expect(results[i].message).toBe(expectedMsg);
+        expect(results[i].severity).toBe(expectedSeverity);
+        expect(results[i].path.join('.')).toBe(expectedPaths[i]);
+      }
+    });
+
+    it('schema example without $ref', async () => {
+      const testDocument = makeCopy(rootDocument);
+
+      testDocument.components.schemas.Drink.example = {
+        $ref: '#/components/schemas/Error',
+      };
+
+      const expectedPaths = ['components.schemas.Drink.example.$ref'];
+
+      const results = await testRule(ruleId, rule, testDocument);
+      expect(results).toHaveLength(1);
+      for (const i in results) {
+        expect(results[i].code).toBe(ruleId);
+        expect(results[i].message).toBe(expectedMsg);
+        expect(results[i].severity).toBe(expectedSeverity);
+        expect(results[i].path.join('.')).toBe(expectedPaths[i]);
+      }
+    });
+
+    it('parameter example with $ref (schema)', async () => {
+      const testDocument = makeCopy(rootDocument);
+
+      testDocument.paths['/v1/drinks'].get.parameters[0].example = {
+        $ref: '#/components/schemas/Error',
+      };
+
+      const expectedPaths = ['paths./v1/drinks.get.parameters.0.example.$ref'];
+
+      const results = await testRule(ruleId, rule, testDocument);
+      expect(results).toHaveLength(1);
+      for (const i in results) {
+        expect(results[i].code).toBe(ruleId);
+        expect(results[i].message).toBe(expectedMsg);
+        expect(results[i].severity).toBe(expectedSeverity);
+        expect(results[i].path.join('.')).toBe(expectedPaths[i]);
+      }
+    });
+
+    it('parameter example with $ref (content)', async () => {
+      const testDocument = makeCopy(rootDocument);
+
+      testDocument.paths['/v1/drinks'].get.parameters[0] = {
+        in: 'query',
+        name: 'start',
+        content: {
+          'application/json': {
+            schema: {
+              type: 'string',
+            },
+            example: {
+              $ref: '#/components/schemas/Error',
```

Comment:
Should we also have a few test cases with more deeply nested ref's?
E..g.:
```yaml
example:
  library:
    name: my-library
    address:
      $ref: '#/components/schemas/Address/example'
    books:
    - $ref: '#/components/schemas/Book/example'
```

Reply:
This is tricky because I don't think that's _technically_ invalid. It's just probably not what the user wants.

For example, the above _could_ be a representation of this valid JSON object:
```json
"library": {
  "name": "my-library",
  "address": {
    "$ref": "#/components/schemas/Address/example"
  },
  "books": [{ "$ref": "#/components/schemas/Book/example" }]
}
```

where `"$ref"` is just an arbitrary key with no meaning. I know that seems rare and maybe we could warn to remind the user its _likely_ wrong but it's hard to tell the difference between unintentional and intentional use.

Reply:
@HenryGessau I can understand the reason for your comment, and this scenario certainly has some subtleties associated with it.  But, I agree with Dustin here in that an argument could be made that the example object you have posted is probably valid.  I think the rule should be focused on looking for a top-level "$ref" property within the "example" object, and avoid checking for nested uses of "$ref" since those might be valid.

Reply:
FWIW, our hacky check found several places in the VPC spec where spec authors have done exactly what Henry's asking about: https://github.ibm.com/riaas/api-spec/pull/6260/files

Reply:
Ok, I'm convinced :joy:
I'll expand the scope of the rule so that it flags a `$ref` property within the `example` field at any level of nesting.

Reply:
fixed